### PR TITLE
✨ #131 support of child gestures over onTargetTap

### PR DESCRIPTION
**Pull request by HappyMakadiyaS**

### CODE REVIEW

File: lib/src/showcase.dart [hunk 1]
```diff
@@ -295,13 +300,6 @@ class _ShowcaseState extends State<Showcase> {
                 ),
               ),
               if (_isScrollRunning) Center(child: widget.scrollLoadingWidget),
-              if (!_isScrollRunning)
-                _TargetWidget(
```
Comment (vatsaltanna-simformsolutions) [inline lib/src/showcase.dart:299]: what is the reason for removing this widget ?

File: lib/src/showcase.dart [hunk 2]
```diff
@@ -327,50 +325,3 @@ class _ShowcaseState extends State<Showcase> {
         : SizedBox.shrink();
   }
 }
-
-class _TargetWidget extends StatelessWidget {
```
Comment (vatsaltanna-simformsolutions) [inline lib/src/showcase.dart:331]: what is the reason for removing this widget ?

Comment (HappyMakadiyaS) [reply lib/src/showcase.dart:331]: The use of _TargetWidget is only to create a container having size of the child and perform onTargetTap action on that container.

This Widget is only used in one place which is no longer needed. ( at 298th line)

File: lib/src/showcase.dart (hunk 1 above)
Comment (HappyMakadiyaS) [reply lib/src/showcase.dart:299]: This widget is used in the stack so the child gesture of overlay is not detectable over _TargetWidget's gesture.
So I have to use the onTargetTap action in the child of the AnchoredOverlay widget of the build method.

**Comment by ParthBaraiya** [comment]
@HappyMakadiyaS some checks are failed in this PR resolve that and update PR.

File: lib/src/showcase.dart [hunk 3]
```diff
@@ -210,7 +210,12 @@ class _ShowcaseState extends State<Showcase> {
         return buildOverlayOnTarget(offset, rectBound.size, rectBound, size);
       },
       showOverlay: true,
-      child: widget.child,
+      child:_showShowCase && widget.onTargetClick != null
+          ? GestureDetector(
+              onTap: _getOnTargetTap,
```
Comment (ParthBaraiya) [inline lib/src/showcase.dart:215]: Provide option for other gestures like, onLongPress and onDoubleTap.

**Comment by PRBaraiya** [comment]
@HappyMakadiyaS  onTargetDoubleTap and onTargetLongPress doesn't work unless I provide onTargetClick.

**Review by ParthBaraiya — APPROVED** [review]

File: lib/src/showcase.dart (hunk 2 above)
Comment (vatsaltanna) [reply lib/src/showcase.dart:331]: please consider to include this widget as it provider gesture to overlay padding.

**Review by vatsaltanna — APPROVED** [review]

**Pull request merged**